worlds/Files.py

```diff
@@ -38,6 +39,32 @@ def get_handler(file: str) -> Optional[AutoPatchRegister]:
         return None
 
 
+class AutoPatchExtensionRegister(abc.ABCMeta):
+    extension_types: ClassVar[Dict[str, AutoPatchExtensionRegister]] = {}
+    required_extensions: List[str] = []
+
+    def __new__(mcs, name: str, bases: Tuple[type, ...], dct: Dict[str, Any]) -> AutoPatchExtensionRegister:
+        # construct class
+        new_class = super().__new__(mcs, name, bases, dct)
+        if "game" in dct:
+            AutoPatchExtensionRegister.extension_types[dct["game"]] = new_class
+        return new_class
+
+    @staticmethod
+    def get_handler(game: str) -> Union[AutoPatchExtensionRegister, List[AutoPatchExtensionRegister]]:
+        handler = AutoPatchExtensionRegister.extension_types.get(game, APPatchExtension)
+        if handler.required_extensions:
+            handlers = [handler]
+            for required in handler.required_extensions:
+                if required in AutoPatchExtensionRegister.extension_types:
+                    handlers.append(AutoPatchExtensionRegister.extension_types.get(required))
+                else:
+                    raise NotImplementedError(f"No handler for {required}.")
```

Comment:
```suggestion
            for required in handler.required_extensions:
                ext = AutoPatchExtensionRegister.extension_types.get(required)
                if not ext:
                    raise NotImplementedError(f"No handler for {required}.")
                handlers.append(ext)
```
We don't need to check before `dict.get()` if `dict.get()` checks.